### ISSUE TITLE
Incorrect refactoring combined hoist-statement-from-loop and aug-assign

### ISSUE THREAD

**Issue by nathanaelhuffman** [sourcery-ai/sourcery#65]
### Issue description or question
Code before the refactor:
```
def crc16(inbytes):
    crc = 0
    for byte in inbytes:
        crc = crc ^ byte << 8
        for _ in range(8):
            if crc & 0x8000:
                crc = crc << 1 ^ 0x1021
            else:
                crc <<= 1
    return crc & 0xFFFF
```
Suggested refactored code, which lost the internal loop making the functionality different.
```
def crc16(inbytes):
    crc = 0
    for byte in inbytes:
        crc ^= byte << 8
        if crc & 0x8000:
            crc = crc << 1 ^ 0x1021
        else:
            crc <<= 1
    return crc & 0xFFFF
```

### Sourcery Version
v0.7.6
Often available in Sourcery settings

### Code editor or IDE name and version

vscode

### OS name and version

Windows

**Comment by Hellebore** [comment]
Thanks for the report. Definitely a bug - have reproduced it.

**Comment by Hellebore** [comment]
Fix identified - will be in the next release.

**Comment by Hellebore** [comment]
resolved with release of 0.7.9
